src/sensor_device.cc

```diff
@@ -57,25 +57,39 @@ void Sensor::update (nut::TcpClient &conn)
 
         _contacts.clear();
 
-        std::string state = nutDevice.getVariableValue (prefix + "contacts.1.status")[0];
-        if (state != "unknown" && state != "bad")
-            _contacts.push_back (state);
-        else
-            log_debug ("sa: %scontact.1.status state %s", prefix.c_str (), state.c_str ());
-
-        state = nutDevice.getVariableValue (prefix + "contacts.2.status")[0];
-        if (state != "unknown" && state != "bad")
-            _contacts.push_back (state);
-        else
-            log_debug ("sa: %scontact.2.status state %s", prefix.c_str (), state.c_str ());
-
-        log_debug ("sa: %scontact.status on %s: contact.1 %s, contact.2 %s",
-                   prefix.c_str (),
-                   location().c_str (),
-                   nutDevice.getVariableValue (prefix + "contacts.1.status")[0].c_str (),
-                   nutDevice.getVariableValue (prefix + "contacts.2.status")[0].c_str ()
-        );
-
+        for (int i = 1 ; i <= 2 ; i++) {
+            std::string baseVar = prefix + "contacts." + std::to_string(i);
+            std::string state = nutDevice.getVariableValue (baseVar + ".status")[0];
+            if (state != "unknown" && state != "bad") {
+                // process new status style (active / inactive), found on EMP002
+                // WRT the polarity configured
+                if (state == "active" || state == "inactive") {
+                    std::string contactConfig = nutDevice.getVariableValue (baseVar + ".config")[0];
+                    if (!contactConfig.empty()) {
+                        if (contactConfig == "normal-opened") {
+                            if (state == "active")
+                                state = "closed";
+                            else
+                                state = "opened";
+                        }
+                        else {
+                            if (state == "active")
+                                state = "opened";
+                            else
+                                state = "closed";
+                        }
+                    }
+                    else {
+                        // FIXME: what to do here? break or?
+                        log_debug ("sa: new style dry-contact status, but missing config");
+                    }
+                }
+                _contacts.push_back (state);
+                log_debug ("sa: %scontact.%i.status state %s", prefix.c_str (), i, state.c_str ());
+            }
+            else
+                log_debug ("sa: %scontact.%i.status state '%s' not supported and discarded", prefix.c_str (), i, state.c_str ());
```

Comment:
Is there a chance to get invalid states that are not strings `unknown` or `bad`? e.g. can there be an empty output of `nutDevice.getVariableValue (baseVar + ".status")[0];` above?